### PR TITLE
Arcane magic rework (part1)

### PR DESCRIPTION
**Pull request by Z0w0plankton**
## About The Pull Request
This PR adds a point-based spell buying system for arcane magic casters. The amount of points is determined by the arcane skill, and possible bonusses based on class/age. Several spells now also award XP when cast, allowing a player to choose more spells as they round continues and they train.

As it stands, I havent added any extra spells past this framework, so the choices for mages remain roughly the same as before, but If this does get approved, I plan to add a number of spells to expand the functionality. 

spell list:
a short destructible forcewall,
a 3x3 AOE that immobilizes for a few seconds if a mob doesnt get out of it in time
a push spell, that flings things directly in front and diagonally from the caster
a message spell, that allows you to message a mob you know (and can remember the name of)
a long-duration AOE damage spell
a single tile damage spellw ith a short delay
a jump spell, allows vertical jumps, and increases spells

LEARNABLE spell list and price:

Greater fireball: 	5
fireball : 		        2
spitfire (fire bolt):	2
fetch:			1
forcewall:		        1
slowdown AOE:	1
message:		        1
push spell:		1
longjump:		1
blade burst:		1
arcyne storm:		2
conjure wolf:		1

Also added an scroll that increases a persons spell points by one, intended for use in dungeon loot

## Why It's Good For The Game
Adds more depth to the magic system, as well as giving the arcane skill more of a use. The hope as that this, along with future spells, will give a bit of variety, choice, and a sense of progression to mages.

### CODE REVIEW

**Comment by Jul-Jul** [comment]
Yeah agree with the previous comment. Love the PR, but without a selection of spells to pick from you're going to get people dumping all their points into fireball.

**Comment by Z0w0plankton** [comment]
That seems like a fair objection. I was planning on making a decent number of spells to go along with this, although I may make those part of this PR directly, instead of a new one later down the line.

**Comment by Tree445** [comment]
Rather than a point system, I think it would be better to force interaction between teacher and student to learn spells. A court mage would be ideal for teaching others spells, and then other people can teach what they know, and likely restrict how many spells you can learn based on arcane level.

**Comment by Jul-Jul** [comment]
A teacher/student system would work if there was more then one master mage (court mage) but forcing people into interaction sounds tedious and not quite fun.

**Comment by Z0w0plankton** [comment]
I think a teaching system, or some other mechanic to learn new spells (that involves other players) might not be a bad idea over a point system. However, I would still like mages to be able to choose a variety of spells when spawning, to give some choice and make each one feel unique. 

On top of that, restricting the knowledge of every spells behind a single role does seem like it could cause frustration, when that role is dead/unfilled/etc.

I feel like a point system is a pretty solid backbone, but perhaps adding some sort of knowledge mechanic ontop of it might be helpful? So a player can spawn with points to buy spells, and knowledge of X spells which can be learned, and copied down onto paper to share. This could lead to some researching mechanic as well, possibly. 

That's just one idea, though, and the first thing I just came up with. There is a discussion thread on the discord if you want to go back and forth a bit more over this?

**Comment by Z0w0plankton** [comment]
This is nearing a point where I'd be able fine with opening it, currently it is just the point system, with some additional spells. I've yet to implement any kind of learning system. Ill probably open this as a proper PR in a day or two.

File: modular_hearthstone/code/modules/spells/roguetown/wizard.dm [hunk 1]
```diff
@@ -174,6 +175,397 @@
...
+		if(SOUTH)
+			lower_left = locate(user.x - 1, user.y - 1, user.z)
+			upper_right = locate(user.x + 1, user.y - 1, user.z)
+		if(WEST)
+			lower_left = locate(user.x - 1, user.y - 1, user.z)
+			upper_right = locate(user.x - 1, user.y + 1, user.z)
+		if(EAST)
+			lower_left = locate(user.x + 1, user.y - 1, user.z)
+			upper_right = locate(user.x + 1, user.y + 1, user.z)
+	return TRUE
```
Comment (WatermelonsEverywhere) [inline modular_hearthstone/code/modules/spells/roguetown/wizard.dm:417]: You should probably be aware Repulse doesn't work since it returns early

**Comment by Tree445** [comment]
Is there any limitations to what spells can be learned? Is there a list?

**Comment by Jul-Jul** [comment]
Yeah be good if you provided a full list of spells that could be learned. =D

File: modular_hearthstone/code/modules/spells/roguetown/wizard.dm (hunk 1 above)
Comment (Z0w0plankton) [reply modular_hearthstone/code/modules/spells/roguetown/wizard.dm]: Fixed, cast needs to return TRUE at the end and I was tired when i added that yesterday. Its in the right place now, tested and repulse works again.

**Comment by Z0w0plankton** [comment]
List of spells added to the PR, including cost

**Comment by Tree445** [comment]
You deleted the pestra magician?

**Pull request merged**